src/azure-cli-core/azure/cli/core/_session.py

```diff
@@ -46,9 +47,16 @@ def load(self, filename, max_age=0):
                     self.save()
             with codecs_open(self.filename, 'r', encoding=self._encoding) as f:
                 self.data = json.load(f)
-        except (OSError, IOError, t_JSONDecodeError):
-            get_logger(__name__).warning("Fail to load or parse file %s. It is overridden by default settings.",
-                                         self.filename)
+        except (OSError, IOError, t_JSONDecodeError) as load_exception:
+            # OSError / IOError should imply file not found issues which are expected on fresh runs (e.g. on build agents
+            # or new systems). A parse error indicates invalid/bad data in the file. We do not wish to warn on missing
+            # files since we expect that, but do if the data isn't as expected.
+            log_level = logger.INFO
+            if isinstance(load_exception, t_JSONDecodeError):
+                log_level = logger.WARNING
+
+            get_logger(__name__).log(log_level, "Failed to load or parse file %s. It will be overridden by default settings.",
+                                     self.filename)
```

Comment:
Do you think if this is better
``` Python
        except (OSError, IOError) as load_exception:
            # OSError / IOError should imply file not found issues which are expected on fresh runs (e.g. on build agents
            # or new systems). A parse error indicates invalid/bad data in the file. We do not wish to warn on missing
            # files since we expect that, but do if the data isn't as expected.
            get_logger(__name__).info("Failed to load %s. It will be overridden by default settings.", self.filename)
        except t_JSONDecoderError:
            get_logger(__name__).warning("Failed to parse the file %s. It will be overridden by default settings.", self.filename)
```

Reply:
I did except you want to note the self.save() at the bottom of exception handling. I wanted to combine them because there is at least some chance in future this block could grow to add more things that would need to be duplicated across both handlers. This mechanism seemed more forward looking.

Also wrote this all in the github editor, so it's a bit error-y, oops!